packages/@jsii/benchmarks/lib/benchmark.ts

```diff
@@ -0,0 +1,134 @@
+import { performance, PerformanceObserver, PerformanceEntry } from 'perf_hooks';
+
+/**
+ * Result of a benchmark run
+ */
+interface Result {
+  /**
+   * The name of the benchmark
+   */
+  name: string;
+
+  /**
+   * The average length across all iterations
+   */
+  average: number;
+
+  /**
+   * Results of individual runs
+   */
+  iterations: PerformanceEntry[];
+}
+
+/**
+ * A simple benchmark for measuring synchronous functions. Uses the `perf_hooks`
+ * module to measure how long a subject takes to execute and averages the result
+ * over all runs. Runs `setup`, `beforeEach`, `afterEach`, and `teardown`
+ * lifecycle hooks before, between, and after runs. These functions, and the
+ * subject function, have access to an optionally defined `context` object that
+ * can be returned from the `setup` function. This allows referencing shared
+ * state across benchmark runs and lifecycle hooks to do things like setup,
+ * teardown, stubbing, etc.
+ */
+export class Benchmark<C> {
+  /**
+   * How many times to run the subject
+   */
+  private _iterations = 5;
+
+  /**
+   * Results of individual runs
+   */
+  private results: PerformanceEntry[] = [];
```

Comment:
Why is this one not using an `_` prefix, when every other private is?

At this rate, couldn't you just use `#private` fields?